pkg/apiserver/validator_test.go

```diff
@@ -113,13 +133,15 @@ func TestValidator(t *testing.T) {
 	if err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
-	status := []ServerStatus{}
-	err = json.Unmarshal(data, &status)
-	if err != nil {
+	var status []ServerStatus
+	if err := json.Unmarshal(data, &status); err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
 	components := util.StringSet{}
 	for _, s := range status {
+		if s.Err != "nil" {
```

Comment:
By the way, this test was passing before when it should have been failing because we weren't checking for error in the sub-components. The test would pass after 30 seconds (hitting the url fetch timeout) because it had a result for each component -- but the result was a failure.